packages/shared/src/components/MainLayout.tsx

```diff
@@ -58,30 +57,30 @@ export default function MainLayout({
   enableSearch,
   onShowDndClick,
 }: MainLayoutProps): ReactElement {
-  const { windowLoaded } = useContext(ProgressiveEnhancementContext);
   const { user, showLogin, loadingUser } = useContext(AuthContext);
   const [showGreeting, setShowGreeting] = useState(false);
   const { sidebarRendered } = useSidebarRendered();
   const [openMobileSidebar, setOpenMobileSidebar] = useState(false);
 
-  return (
-    <>
-      <PromotionalBanner />
-      <header className="flex relative laptop:fixed laptop:top-0 laptop:left-0 z-3 flex-row-reverse laptop:flex-row justify-between items-center py-3 px-4 tablet:px-8 laptop:px-4 laptop:w-full h-14 border-b bg-theme-bg-primary border-theme-divider-tertiary">
-        {!sidebarRendered && <MobileHeaderRankProgress />}
-        {mobileTitle && (
-          <h3 className="block laptop:hidden typo-callout">{mobileTitle}</h3>
-        )}
-        {shouldShowLogo({ mobileTitle, sidebarRendered }) && (
-          <Logo onLogoClick={onLogoClick} showGreeting={showGreeting} />
-        )}
-        {windowLoaded && greeting && (
+  const LogoAndGreeting = () => {
```

Comment:
Let's take it outside this component. (you can keep it in the same file)
If you need you can put its own state inside as well